.github/workflows/create-tfe-release-notes.yml

```diff
@@ -0,0 +1,108 @@
+name: Create TFE Release Notes
+run-name: Create changelog PR for TFE ${{ inputs.version }} ${{ inputs.dev-mode && 'in dev mode' || '' }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'The TFE version for the upcoming TFE release, format is either vYYYYMM-# or MAJOR.MINOR.PATCH (without a "v" prefix).'
+        required: true
+      release-branch:
+        description: 'The release branch for the upcoming TFE release, format is either ptfe-releases/vYYYYMM or release/MAJOR.MINOR.x (without a "v" prefix).'
+        required: true
+      last-release-tag:
+        description: 'The Github Tag for the TFE release to generate a changelog from, format is either vYYYYMM-# or vMAJOR.MINOR.PATCH (with a "v" prefix).'
+        required: true
+      dev-mode:
+        description: 'Whether to run in dev mode, a flag for creating the release notes changelog PR'
+        type: boolean
+        required: false
+        default: false
+
+env:
+  GITHUB_TOKEN: ${{ secrets.TFE_GITHUB_TOKEN_010623 }}
+  GITHUB_USER: ${{ secrets.TFE_GITHUB_USER_010623 }}
+  DEV: ${{ inputs.dev-mode }}
+  VERSION: ${{ inputs.version }}
+  RELEASE_BRANCH: ${{ inputs.release-branch }}
+  LAST_RELEASE_TAG: ${{ inputs.last-release-tag }}
+
+jobs:
+    release-notes:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Only run in hashicorp/web-unified-docs-internal
+          run: |
+            if [ "${{ github.repository }}" != "hashicorp/web-unified-docs-internal" ]; then
+              echo "This workflow workflow will only work in hashicorp/web-unified-docs-internal"
+              exit 1
+            fi
+
+        - name: Checkout web-unified-docs repository
+          uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+          with:
+            # !! remove this once approved !!
+            ref: rn/add-ptfe-release-workflow
+
+        - name: Install and cache Ruby gems at root
+          uses: ruby/setup-ruby@52753b7da854d5c07df37391a986c76ab4615999 # v1.191.0
+          with:
+            bundler-cache: true
+
+        - name: Clone Services Repos for Changelog Script
+          run: scripts/tfe-releases/ci/clone-service-repos.sh
+
+        - name: Run execute-changelog-script.sh
+          run: scripts/tfe-releases/ci/execute-changelog-script.sh
+
+        - name: Turn branch into PR
+          run: |
+            PR_URL=$(scripts/tfe-releases/ci/create-pull-request.sh)
+            echo "PR_URL=$PR_URL" >> $GITHUB_ENV
+
+        - name: Create PR URL
+          if: ${{ env.DEV == 'false' }}
+          run: |
+            echo "[TFE release notes PR for ${{ env.VERSION }}](${{ env.PR_URL }})" >> $GITHUB_STEP_SUMMARY
+
+        # - name: Slack notification
```

Comment:
Commented out so that no one gets spammed in slack. Will remove comment once shipped.